modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/RouterModelLoaderDelegate.java

```diff
@@ -47,7 +49,7 @@ void declareRouter(ExtensionDeclarer extensionDeclarer, HasConstructDeclarer own
     parser.getAdditionalModelProperties().forEach(router::withModelProperty);
 
     loader.getParameterModelsLoaderDelegate().declare(router, parser.getParameterGroupModelParsers());
-
+    addSemanticTerms(router.getDeclaration(), parser);
```

Comment:
does this also needs to be added to its nested routes?

Reply:
no